src/test/java/com/openlattice/rehearsal/entitysets/EntitySetsTest.kt

```diff
@@ -63,4 +73,133 @@ class EntitySetsTest : MultipleAuthenticatedUsersBase() {
                 "Linked entity sets is empty"
         )
     }
+
+    @Test
+    fun testCreateBadDataExpiration() {
+        //set expiration policy with negative duration of time until data expires
+        val badTTL = -10L
+        assertException({ DataExpiration(badTTL, ExpirationBase.LAST_WRITE, DeleteType.Hard) },
+                "Time until data expiration must not be negative")
+
+        //set expiration policy without required start date
+        val tTL = 10L
+        assertException({ DataExpiration(tTL, ExpirationBase.DATE_PROPERTY, DeleteType.Soft) },
+                "Must provide property type for expiration calculation")
+    }
+
+    @Test
+    fun testAddExpirationPolicy() {
+        val es = createEntitySet()
+        Assert.assertNull(es.expiration) // default entity set has no expiration policy
+
+        //set expiration policy
+        val tTL = 10L
+        val expirationPolicy = DataExpiration(tTL, ExpirationBase.FIRST_WRITE, DeleteType.Hard)
+        val update = MetadataUpdate(Optional.empty(), Optional.empty(), Optional.empty(),
```

Comment:
since there's a lot of these expirationPolicy MetadataUpdates with about a million Optional.empty()s, you could add a helper function to just generate and return that object (and then if the MetadataObject ever changes in the future, it'll be much easier to update here)